pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -341,6 +352,13 @@ private void computeKeepIndexing() {
         keepIndexing = false;
       }
     }
+    updateCurrentDocumentCountMetrics();
```

Comment:
Not here, I would do it after we finish creating the index segment (but before we call notifySegmentCommitted)